lnwallet/btcwallet/btcwallet_test.go

```diff
@@ -0,0 +1,168 @@
+package btcwallet
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/btcsuite/btcd/wire"
+	"github.com/btcsuite/btcwallet/wallet"
+)
+
+type previousOutpoint struct {
+	outpoint    string
+	isOurOutput bool
+}
+
+type previousOutpointsTest struct {
+	name     string
+	tx       *wire.MsgTx
+	myInputs []wallet.TransactionSummaryInput
+	expRes   []previousOutpoint
+}
+
+var previousOutpointsTests = []previousOutpointsTest{
+	{
+		name: "both outpoints are wallet controlled",
+		tx: &wire.MsgTx{
+			TxIn: []*wire.TxIn{
+				{
+					PreviousOutPoint: wire.OutPoint{
+						Index: 0,
+					},
+				},
+				{
+					PreviousOutPoint: wire.OutPoint{
+						Index: 1,
+					},
+				},
+			},
+		},
+		myInputs: []wallet.TransactionSummaryInput{
+			{
+				Index: 0,
+			},
+			{
+				Index: 1,
+			},
+		},
+		expRes: []previousOutpoint{
+			{
+				outpoint: wire.OutPoint{
+					Index: 0,
+				}.String(),
+				isOurOutput: true,
+			},
+			{
+				outpoint: wire.OutPoint{
+					Index: 1,
+				}.String(),
+				isOurOutput: true,
+			},
+		},
+	},
+	{
+		name: "only one outpoint is wallet controlled",
+		tx: &wire.MsgTx{
+			TxIn: []*wire.TxIn{
+				{
+					PreviousOutPoint: wire.OutPoint{
+						Index: 0,
+					},
+				},
+				{
+					PreviousOutPoint: wire.OutPoint{
+						Index: 1,
+					},
+				},
+			},
+		},
+		myInputs: []wallet.TransactionSummaryInput{
+			{
+				Index: 0,
+			},
+			{
+				Index: 2,
+			},
+		},
+		expRes: []previousOutpoint{
+			{
+				outpoint: wire.OutPoint{
+					Index: 0,
+				}.String(),
+				isOurOutput: true,
+			},
+			{
+				outpoint: wire.OutPoint{
+					Index: 1,
+				}.String(),
+				isOurOutput: false,
+			},
+		},
+	},
+	{
+		name: "no outpoint is wallet controlled",
+		tx: &wire.MsgTx{
+			TxIn: []*wire.TxIn{
+				{
+					PreviousOutPoint: wire.OutPoint{
+						Index: 0,
+					},
+				},
+				{
+					PreviousOutPoint: wire.OutPoint{
+						Index: 1,
+					},
+				},
+			},
+		},
+		myInputs: []wallet.TransactionSummaryInput{
+			{
+				Index: 2,
+			},
+			{
+				Index: 3,
+			},
+		},
+		expRes: []previousOutpoint{
+			{
+				outpoint: wire.OutPoint{
+					Index: 0,
+				}.String(),
+				isOurOutput: false,
+			},
+			{
+				outpoint: wire.OutPoint{
+					Index: 1,
+				}.String(),
+				isOurOutput: false,
+			},
+		},
+	},
+}
+
+// TestPreviousOutpoints tests if we are able to get the previous
+// outpoints correctly.
+func TestPreviousOutpoints(t *testing.T) {
+	for _, test := range previousOutpointsTests {
+		t.Run(test.name, func(t *testing.T) {
+			respOutpoints := getPreviousOutpoints(test.tx, test.myInputs)
+
+			for idx, respOutpoint := range respOutpoints {
+				expRes := test.expRes[idx]
+				if !reflect.DeepEqual(expRes.outpoint,
```

Comment:
New code should use the `require` library. I think the `require.Equal()` should be exactly what you need here.